infinispan/src/main/java/org/hibernate/search/infinispan/impl/InfinispanDirectoryProvider.java

```diff
@@ -129,8 +136,30 @@ public void start(DirectoryBasedIndexManager indexManager) {
 		log.debugf( "Initialized Infinispan index: '%s'", directoryProviderName );
 	}
 
+	private void augmentDirectoryWithAsyncDeletes(org.infinispan.lucene.directory.BuildContext directoryBuildContext) {
+		Executor executor = deletesExecutor.getExecutor();
+		try {
+			Method operationsExecutor = directoryBuildContext.getClass().getMethod(
+					"deleteOperationsExecutor",
+					Executor.class
+			);
+			operationsExecutor.invoke( directoryBuildContext, executor );
+		}
+		catch (NoSuchMethodException e) {
+			log.notSettingAsyncDeletes();
+		}
+		catch (InvocationTargetException e) {
+			log.notSettingAsyncDeletes();
```

Comment:
Let's assume we get the InvocationTargetException (or the next one) triggered. Are you sure you wouldn't love to be able to tell the reason is different than the first case? I think I'd add three different log methods, or a different one for the 2nd and 3rd case using a Cause annotation.